### PR TITLE
DB and Router Setup 

### PR DESCRIPTION
**Pull request by seancorc**
* Create user model
* Setup connection to DB
* Setup router
* Add endpoint for creating a user

### CODE REVIEW

File: src/appdev/ApplicationRouter.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
+import { NextFunction, Request, Response, Router } from 'express';
+
+/**
+ * RequestType - the HTTP methods supported by ApplicationRouter
+ */
+export type RequestType = 'GET' | 'POST' | 'DELETE';
```
Comment (orkosinha) [inline src/appdev/ApplicationRouter.ts:6]: Are we supposed to use these in routers?

File: src/API.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import bodyParser from 'body-parser';
+import { Router } from 'express';
+import ApplicationAPI from './appdev/ApplicationAPI';
+import CreateUserRouter from './routers/CreateUserRouter'
```
Comment (leewang-codes) [inline src/API.ts]: keep consistent `;` endings

**Review by orkosinha — APPROVED** [review]
Looks 🔥

File: src/appdev/ApplicationRouter.ts (hunk 1 above)
Comment (seancorc) [reply src/appdev/ApplicationRouter.ts:6]: Ya, they're used in the ApplicationRouter's constructor.

File: src/entities/User.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+@Entity('users')
+class User {
+
+  @PrimaryGeneratedColumn('uuid')
+  id: string;
+
+  /** User first name */
+  @Column('character varying')
+  firstName: string = '';
```
Comment (leewang-codes) [inline src/entities/User.ts]: i think newer versions of typeorm prefer setting default values like this instead:
```
@Column({
  type: 'character varying',
  default: ''
})
firstName: string;
```

see https://typeorm.io/#/entities/column-options

File: src/repos/UserRepo.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+const createUser = async (
+  netID: string,
+  googleID: string,
+  firstName: string,
+  lastName: string,
+): Promise<User> => {
+  try {
+    if (!(netID && googleID && firstName && lastName)) {
+      throw Error();
```
Comment (leewang-codes) [inline src/repos/UserRepo.ts]: add error message

File: src/routers/CreateUserRouter.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    super('POST');
+  }
+
+  getPath(): string {
+    return '/user/';
+  }
+
+  async content(req: Request): Promise<SerializedUser> {
+    const { firstName, googleID, lastName, netID } = req.body;
+    const user = await UserRepo.createUser(netID, googleID, firstName, lastName)
```
Comment (leewang-codes) [inline src/routers/CreateUserRouter.ts]: be consistent with `;`

**Review by leewang-codes — CHANGES_REQUESTED** [review]
great start - just some small things to fix

File: .envrctemplate [hunk 6]
```diff
@@ -0,0 +1,8 @@
+export PORT=[port]
+export DB_PORT=[db port]
+export HOST=[host]
+export DB_USERNAME=[db username]
+export DB_PASS=[db pass]
+export DB_NAME=[db name]
+export NODE_ENV=[node env]
```
Comment (alanna-zhou) [inline .envrctemplate:6]: alphabetize?! lol

**Review by alanna-zhou — APPROVED** [review]
looks good what a star

**Review by leewang-codes — APPROVED** [review]

**Pull request merged**